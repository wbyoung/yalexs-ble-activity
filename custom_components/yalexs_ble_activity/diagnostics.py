"""Diagnostics support for Yale Access Bluetooth Activity."""

from typing import Any

from homeassistant.components.diagnostics import async_redact_data
from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant

TO_REDACT: set[str] = set()


async def async_get_config_entry_diagnostics(  # noqa: RUF029
    hass: HomeAssistant,  # noqa: ARG001
    entry: ConfigEntry,
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    result: dict[str, Any] = async_redact_data(entry.as_dict(), TO_REDACT)
    return result
