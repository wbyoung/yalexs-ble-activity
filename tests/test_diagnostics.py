"""Test Yale Access Bluetooth Activity diagnostics."""

from homeassistant.core import HomeAssistant
from pytest_homeassistant_custom_component.common import MockConfigEntry
from pytest_homeassistant_custom_component.components.diagnostics import (
    get_diagnostics_for_config_entry,
)
from pytest_homeassistant_custom_component.typing import ClientSessionGenerator
from syrupy.assertion import SnapshotAssertion
from syrupy.filters import props

from . import setup_integration


async def test_entry_diagnostics(
    hass: HomeAssistant,
    hass_client: ClientSessionGenerator,
    config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test config entry diagnostics."""
    await setup_integration(hass, config_entry)

    assert await get_diagnostics_for_config_entry(
        hass, hass_client, config_entry
    ) == snapshot(
        exclude=props(
            "id",
            "device_id",
            "entry_id",
            "created_at",
            "modified_at",
            "expires_at",
            "config_entries",
            "config_entries_subentries",
            "primary_config_entry",
        )
    )
