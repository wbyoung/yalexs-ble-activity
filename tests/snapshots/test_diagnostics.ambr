# serializer version: 1
# name: test_entry_diagnostics
  dict({
    'data': dict({
      'lock_entities': list([
        'lock.front_door',
      ]),
    }),
    'disabled_by': None,
    'discovery_keys': dict({
    }),
    'domain': 'yalexs_ble_activity',
    'minor_version': 1,
    'options': dict({
    }),
    'pref_disable_new_entities': False,
    'pref_disable_polling': False,
    'source': 'user',
    'subentries': list([
    ]),
    'title': 'Yale Access Bluetooth Activity',
    'unique_id': None,
    'version': 1,
  })
# ---
